base/src/main/groovy/org/asciidoctor/gradle/base/AbstractAsciidoctorBaseTask.groovy

```diff
@@ -102,14 +104,22 @@ abstract class AbstractAsciidoctorBaseTask extends DefaultTask {
      * @since 3.0
      */
     void sourceDir(Object f) {
-        this.srcDir = f
+        this.srcDir.set(project.file(f))
```

Comment:
Same here.

Reply:
If you wrap the `project.file` in a provider before passing it to the `.set()` I would be more at ease.

Reply:
Same as above.